core/vm/contracts.go

```diff
@@ -67,21 +67,27 @@ func ripemd160Func(in []byte) []byte {
 const ecRecoverInputLength = 128
 
 func ecrecoverFunc(in []byte) []byte {
-	// "in" is (hash, v, r, s), each 32 bytes
-	// but for ecrecover we want (r, s, v)
 	if len(in) < ecRecoverInputLength {
```

Comment:
Please remove this if statement. `RightPadBytes` makes this obsolete. It will do the check for you